play/src/front/Api/Events/Ui/ButtonActionBarEvent.ts

```diff
@@ -3,6 +3,9 @@ import { z } from "zod";
 export const isAddButtonActionBarEvent = z.object({
     id: z.string(),
     label: z.string(),
+    type: z.enum(["button", "action"]).optional().default("button"),
+    imageSrc: z.string().optional().nullable(),
+    toolTip: z.string().optional().nullable(),
```

Comment:
Here, label is useless for an `action` type. So it should be optional.

But you could technically, create invalid objects.

Like

```
{
  id: "456456",
  label: "foo",
  type: "button"
}
```

The "button" type MUST have an imageSrc.

What we need is discriminated unions.

Using Zod, you can make an union on 2 different types. One with

```
type: z.litteral("button")
```

and the other one with

```
type: z.litteral("action")
```

This way, we can target EXACTLY what is need in both cases.

Also, "button" and "action" are maybe confusing. Both are actually "buttons" that do "actions". It is just that one is an icon and the other a text button.

So what about renaming the type to "textButton" and "iconButton" (or something like that)?